CompanionPane/src/androidTest/java/com/microsoft/device/display/samples/companionpane/LayoutTest.kt

```diff
@@ -0,0 +1,129 @@
+     * the portrait orientation
+     */
+    @Test
+    fun app_testSinglePortraitLayout() {
+        composeTestRule.setContent {
+            CompanionPaneAppTheme {
+                CompanionPaneApp(composeTestRule.activity.rememberWindowState())
+            }
+        }
+
+        // Lock in portrait orientation
+        device.setOrientationNatural()
+        device.freezeRotation()
+
+        // Check that single portrait layout is shown
+        composeTestRule.onNodeWithTag(composeTestRule.getString(R.string.single_port))
+            .assertIsDisplayed()
+
+        // Unfreeze orientation
+        device.unfreezeRotation()
+    }
+
+    /**
+     * Tests that the single landscape layout appears when one pane is shown and the device is in
+     * the landscape orientation
+     */
+    @Test
+    fun app_testSingleLandscapeLayout() {
+        composeTestRule.setContent {
+            CompanionPaneAppTheme {
+                CompanionPaneApp(composeTestRule.activity.rememberWindowState())
+            }
+        }
+
+        // Lock in landscape orientation
+        device.setOrientationLeft()
+        device.freezeRotation()
+
+        // Check that single landscape layout is shown
+        composeTestRule.onNodeWithTag(composeTestRule.getString(R.string.single_land))
+            .assertIsDisplayed()
+
+        // Unfreeze orientation
+        device.unfreezeRotation()
+    }
+
+    /**
+     * Tests that the dual portrait layout appears when a vertical fold is simulated
+     */
+    @ExperimentalTestApi
+    @Test
+    fun app_testDualPortraitLayout() {
+        composeTestRule.setContent {
+            CompanionPaneAppTheme {
+                CompanionPaneApp(composeTestRule.activity.rememberWindowState())
+            }
+        }
+
+        // Simulate vertical fold
+        publisherRule.simulateVerticalFold(composeTestRule)
+
+        // Check that dual portrait panes are shown
+        composeTestRule.onNodeWithTag(composeTestRule.getString(R.string.dual_port_pane1))
+            .assertIsDisplayed()
+        composeTestRule.onNodeWithTag(composeTestRule.getString(R.string.dual_port_pane2))
+            .assertIsDisplayed()
+    }
+
+    /**
+     * Tests that the dual landscape layout appears when a horizontal fold is simulated
+     */
+    @ExperimentalTestApi
+    @Test
+    fun app_testDualLandscapeLayout() {
+        composeTestRule.setContent {
+            CompanionPaneAppTheme {
+                CompanionPaneApp(composeTestRule.activity.rememberWindowState())
+            }
+        }
+
+        // Simulate horizontal fold
+        publisherRule.simulateHorizontalFold(composeTestRule)
+
+        // Check that dual portrait panes are shown
+        composeTestRule.onNodeWithTag(composeTestRule.getString(R.string.dual_land_pane1))
+            .assertIsDisplayed()
+        composeTestRule.onNodeWithTag(composeTestRule.getString(R.string.dual_land_pane2))
+            .assertIsDisplayed()
+    }
+}
```

Comment:
Can we please add some test cases for the device rotating from portrait/dual-portrait to landscape/dual-lanscape?

Reply:
Yes, I can add those now!